tests/unit/type/string.tcl

```diff
@@ -256,6 +256,74 @@ start_server {tags {"string"}} {
         list [r msetnx x1{t} xxx x1{t} zzz] [r get x1{t}]
     } {0 yyy}
 
+    test {MSETEX - all expiration flags} {
+        # Test each expiration type separately (EX, PX, EXAT, PXAT)
+        set future_sec [expr [clock seconds] + 10]
+        set future_ms [expr [clock milliseconds] + 10000]
+
+        # Test EX and PX with separate commands (each applies to all keys in that command)
+        r msetex keys 2 ex:key1{t} val1 ex:key2{t} val2 ex 5
+        r msetex keys 2 px:key1{t} val1 px:key2{t} val2 px 5000
+
+        # Test EXAT and PXAT with separate commands
+        r msetex keys 2 exat:key1{t} val3 exat:key2{t} val4 exat $future_sec
+        r msetex keys 2 pxat:key1{t} val3 pxat:key2{t} val4 pxat $future_ms
+
+        set ex_ttl_ok [expr [r ttl ex:key1{t}] > 0]
+        set px_ttl_ok [expr [r pttl px:key1{t}] > 0]
+        set exat_ttl_ok [expr [r ttl exat:key1{t}] > 0]
+        set pxat_ttl_ok [expr [r pttl pxat:key1{t}] > 0]
+        list $ex_ttl_ok $px_ttl_ok $exat_ttl_ok $pxat_ttl_ok
+    } {1 1 1 1}
+
+    test {MSETEX - KEEPTTL preserves existing TTL} {
+        r setex keepttl:key{t} 100 oldval
+        set old_ttl [r ttl keepttl:key{t}]
+        r msetex keys 1 keepttl:key{t} newval keepttl
+        list [r get keepttl:key{t}] [expr [r ttl keepttl:key{t}] >= [expr $old_ttl - 5]]
+    } {newval 1}
```

Comment:
These assertions are not related. Separating them makes it easier to know which test fails.

```suggestion
        assert_equal [r get keepttl:key{t}] "newval"
        assert_morethan [r ttl keepttl:key{t}] [expr $old_ttl - 5]
    }
```